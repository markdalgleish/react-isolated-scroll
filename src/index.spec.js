import IsolatedScroll from './index'
import ReactTestRenderer from 'react-test-renderer';
import React from 'react'

jest.mock('isolated-scroll')

describe('IsolatedScroll', () => {
  const mockedIsolatedScroll = require('isolated-scroll')
  beforeEach(() => {
    mockedIsolatedScroll.mockReset()
  })
  it('should be a React Component', () => {
    expect(IsolatedScroll).toBeInstanceOf(Function)
  })
  it('should mount isolatedScroll binding', () => {
    ReactTestRenderer.create(<IsolatedScroll />)
    expect(mockedIsolatedScroll).toHaveBeenCalledTimes(1)
  })
  it('should unmount isolatedScroll binding', () => {
    const mockedUnbindHandlersKey = jest.fn()
    mockedIsolatedScroll.mockImplementation(() => mockedUnbindHandlersKey)
    const instance = ReactTestRenderer.create(<IsolatedScroll />)
    instance.unmount()
    expect(mockedUnbindHandlersKey).toHaveBeenCalledTimes(1)
  })
  it('should render', () => {
    const tree = ReactTestRenderer.create(
      <IsolatedScroll />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
  it('should pass through children', () => {
    const tree = ReactTestRenderer.create(
      <IsolatedScroll>exampleChildTextNode</IsolatedScroll>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
  it('should pass through attributes', () => {
    const tree = ReactTestRenderer.create(
      <IsolatedScroll data-example-attribute="exampleValue" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })

})
