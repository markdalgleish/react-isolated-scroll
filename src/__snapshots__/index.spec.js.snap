// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsolatedScroll should pass through attributes 1`] = `
<div
  data-example-attribute="exampleValue"
/>
`;

exports[`IsolatedScroll should pass through children 1`] = `
<div>
  exampleChildTextNode
</div>
`;

exports[`IsolatedScroll should render 1`] = `<div />`;
